mm2src/lp_ordermatch.rs

```diff
@@ -3703,34 +3885,154 @@ fn my_taker_order_file_path(ctx: &MmArc, uuid: &Uuid) -> PathBuf {
     my_taker_orders_dir(ctx).join(format!("{}.json", uuid))
 }
 
+fn my_order_history_file_path(ctx: &MmArc, uuid: &Uuid) -> PathBuf {
+    my_orders_history_dir(ctx).join(format!("{}.json", uuid))
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+struct HistoricalOrder {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    max_base_vol: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    min_base_vol: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    price: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    updated_at: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    conf_settings: Option<OrderConfirmationsSettings>,
+}
+
+impl HistoricalOrder {
+    fn build(new_order: &MakerOrder, old_order: &MakerOrder) -> HistoricalOrder {
```

Comment:
It is possible to build it from `MakerOrderUpdated`? It seems to already contain the required information so there will be no need to clone the old order and then compare all the new and old fields.